### PR TITLE
RS/YJ/Rule 11-16 (include only necessary codes)

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
This PR replaces PR https://github.com/pnnl/ruleset-checking-tool/pull/1545. Thank you!

### CODE REVIEW

File: rct229/rulesets/ashrae9012019/section11/section11rule16.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            each_rule=Section11Rule16.SWHEquipRule(),
+            index_rmd=BASELINE_0,
+            id="11-16",
+            description="Gas water heaters shall be modeled using natural gas as their fuel. Exceptions: Where natural gas is not available for the proposed building site, as determined by the rating authority, gas water heaters shall be modeled using propane as their fuel.",
+            ruleset_section_title="Service Water Heating",
+            standard_section="Table G3.1 #11, baseline column, (h)",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0].service_water_heating_equipment[*]",
+        )
+
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule16.py:33]: Here is missing a `create_data` to get the `proposed_fuels`

```python
def create_data(self, context, data):
    rmd_p = context.PROPOSED
    return {
        "proposed_fuels": find_all("$.ruleset_model_descriptions[0].service_water_heating_equipment[*].heater_fuel_type)
   }
```
You do not need PROPOSED in the `SWHEquipRule` level but simply take out this list.

**Review by weilixu — COMMENTED** [review]
One logic is missing in the implementation

File: rct229/rulesets/ashrae9012019/section11/section11rule16.py (hunk 1 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section11/section11rule16.py:38]: Addressed.

File: rct229/rulesets/ashrae9012019/section11/section11rule16.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    def create_data(self, context, data):
+        rmd_p = context.PROPOSED
+
+        return {
+            "proposed_fuels": find_all(
+                "$.ruleset_model_descriptions[0].service_water_heating_equipment[*].heater_fuel_type",
+                rmd_p,
+            )
+        }
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule16.py:47]: in here, replace with the new function to find all fuels in the proposed model (follows the RDS implementation)

**Review by weilixu — COMMENTED** [review]
Use the new fuel source function

File: rct229/rulesets/ashrae9012019/section11/section11rule16.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section11/section11rule16.py]: Good point. Addressed.

**Review by weilixu — APPROVED** [review]

**Pull request merged**